### PR TITLE
work around ansible bug where command must be called ssh

### LINKED ISSUE #1266: --become-password causes all tasks to timeout
## Specify type:
- Bug

### Priority:
- Blocker

___

## Description:
Using the --become-password (with or without --become-user) causes tasks to timeout and scans to fail.
___

## Acceptance Criteria:

- [ ] Verify that users can can use a regular user with sudo password to run a scan.
___

### Version Information:
0.0.41 possibly master

### Expected behavior:
Describe the expected behavior for the program.

### Actual behavior:
Provide the actual behavior for the program.

### Steps to reproduce:
 qpc cred add --name sudo_test2 --username mvickstr --password --become-password --type network
 qpc source add --type network --name sudo2 --cred sudo_test2 --hosts 192.168.56.0.24
 qpc scan add --name test_sudo2 --sources sudo2

### Environment information:
Provide detailed information on the *scanning* and *target* systems (if applicable); see example below.

Type            | Operating System | OS Version | Python Version | Virtualization
--------------- | ---------------- | ---------- | -------------- | --------------
Scanning System | RHEL             | 7.4        | 3.6            | Bare Metal
Target System   | RHEL             | 5.9        | 2.4.6          | RHEV

### PR DESCRIPTION
**Pull request by kholdaway**
Closes #1266

### CODE REVIEW

File: bin/timeout_ssh [hunk 1]
```diff
@@ -23,6 +23,7 @@ ARG_TIMEOUT = '--timeout='
 # program whose only job is to be a single executable that Ansible can
 # call which will then trampoline out to timeout.
 
+
```
Comment (noahl) [inline bin/timeout_ssh]: Nitpick: extra space here

File: bin/timeout_ssh [hunk 2]
```diff
@@ -51,9 +52,15 @@ def main(argv):
...
+    # the ssh arg is required for become-pass because
+    # ansible checks for an exact string match of ssh
+    # anywhere in the command array
+    # See https://github.com/ansible/ansible/blob/stable-2.3/lib/ansible/plugins/connection/ssh.py#L490-L500
+    # the code below will remove the ssh argument before running the command
     executable_args = [arg for arg in argv[1:]
                        if (not arg.startswith(ARG_EXECUTABLE) and
-                           not arg.startswith(ARG_TIMEOUT))]
+                           not arg.startswith(ARG_TIMEOUT) and
+                           arg != 'ssh')]
```
Comment (noahl) [inline bin/timeout_ssh]: Nit: maybe declare a global variable `SSH = 'ssh'` and then use it here?

Comment (kholdaway) [reply bin/timeout_ssh]: 👍

File: bin/timeout_ssh (hunk 1 above)
Comment (kholdaway) [reply bin/timeout_ssh]: I am going to leave that one because my auto-formatter put that there.

**Pull request merged**